pkg/apis/stork/v1alpha1/clusterdomains.go

```diff
@@ -0,0 +1,75 @@
+package v1alpha1
+
+import (
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// ClusterDomainsStatusResourceName is name for "clusterdomainsstatus" resource
+	ClusterDomainsStatusResourceName = "clusterdomainsstatus"
+	// ClusterDomainsUpdateResourceName is name for "clusterdomainsupdate" resource
+	ClusterDomainsUpdateResourceName = "clusterdomainsupdate"
+	// ClusterDomainsUpdatePlural is plural for "clusterdomainsupdate" resource
+	ClusterDomainsUpdatePlural = "clusterdomainsupdates"
+)
+
+// ClusterDomains provides a list of activated cluster domains and a list
+// of inactive cluster domains
+type ClusterDomains struct {
+	Active   []string `json:"active"`
+	Inactive []string `json:"inactive"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterDomainsStatus represents the status of all cluster domains
+type ClusterDomainsStatus struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	Status          ClusterDomains `json:"status"`
```

Comment:
are end users expected to interact with the status field to set active and inactive clusters?

Reply:
No. The ClusterDomainsStatus CRD, will be automatically applied when stork is enabled with ClusterDomains.
It will be only used to display the current status of all cluster domains.

Reply:
Okay. In that case, will end users interact with `ClusterDomains`.

May be I am missing the workflow for the operation: "ClusterA is down right now. Mark it inactive"

Reply:
The Status CRD will not be editable by the user.

To activate/deactivate a domain, they will have to use the ClusterDomainsUpdate CRD.